### PR TITLE
feat(wallet): ability to log out from loader

### LINKED ISSUE #1989: Ability to cancel wallet connection attempt
When ZAP wallet is setup and connecting (or with wrong settings: not connecting); then there is no ability to change the connections settings; the only solution is to fully uninstall and manually remove data folders (which are left after uninstall).

## Detailed Description
Would be great to have a simple ability to change the connection settings on an already installed ZAP wallet.

### PR DESCRIPTION
**Pull request by mrfelton**
## Description:

Sometimes it happens that a wallet login is successful, but the wallet ui doesn't show due to missing dependencies such as ticker information or required information about the lnd node that we connected to. Provide a means to cancel the failed login attempt and return to the launchpad.

Depends on #1989 

## Motivation and Context:

Fix #1989

## How Has This Been Tested?

Manually.

## Screenshots:

(note X in upper right corner - this will trigger a logout)

image

## Types of changes:

Enhancement

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have reviewed and updated the documentation accordingly.
- [x] I have read the _CONTRIBUTING_ document.
- [ ] I have added tests to cover my changes where needed.
- [ ] All new and existing tests passed.
- [x] My commits have been squashed into a concise set of changes.

### CODE REVIEW

File: renderer/components/LoadingBolt/LoadingBolt.js [hunk 1]
```diff
@@ -48,43 +51,79 @@ const FullPageGradient = styled(Flex)`
...
+const LoadingCloseButtonWrapper = styled(Box)`
+  height: 40px;
+  cursor: pointer;
+  opacity: 0.6;
+  &:hover: {
+    opacity: 1;
+  }
+`
+
+const LoadingCloseButton = ({ onClose }) => (
```
Comment (korhaliv) [inline renderer/components/LoadingBolt/LoadingBolt.js]: We have close button in a couple of places already (for instance in AutopayModalBody). Would be nice to finally extract this and avoid duplicating them.

File: renderer/components/LoadingBolt/LoadingBolt.js [hunk 2]
```diff
@@ -48,43 +51,79 @@ const FullPageGradient = styled(Flex)`
...
+LoadingCloseButton.propTypes = {
+  onClose: PropTypes.func,
+}
+
 class LoadingBolt extends React.PureComponent {
   static propTypes = {
+    hasClose: PropTypes.bool,
     isLoading: PropTypes.bool.isRequired,
     message: PropTypes.object,
+    onClose: PropTypes.func,
```
Comment (korhaliv) [inline renderer/components/LoadingBolt/LoadingBolt.js]: Looks like this prop is not required, however it is directly passed into an onClick without a null check

File: renderer/reducers/app.js [hunk 3]
```diff
@@ -49,6 +52,14 @@ export function resetApp() {
   }
 }
 
+export const logout = () => async dispatch => {
```
Comment (korhaliv) [inline renderer/reducers/app.js]: `await` is not used anywhere inside, nor does this function return a promise. consider removing `async` modifier

File: renderer/components/LoadingBolt/LoadingBolt.js (hunk 1 above)
Comment (mrfelton) [reply renderer/components/LoadingBolt/LoadingBolt.js]: Have extracted this into a new `CloseButton` component.

File: renderer/components/LoadingBolt/LoadingBolt.js (hunk 2 above)
Comment (mrfelton) [reply renderer/components/LoadingBolt/LoadingBolt.js]: Added null handling in `CloseButton`

File: renderer/reducers/app.js (hunk 3 above)
Comment (mrfelton) [reply renderer/reducers/app.js]: Updated

**Comment by mrfelton** [comment]
Superseded by https://github.com/LN-Zap/zap-desktop/pull/2069

**Pull request closed without merging**